src/Cms/Page.php

```diff
@@ -929,26 +930,31 @@ public function permissions(): PagePermissions
 	 * Draft preview Url
 	 * @internal
 	 */
-	public function previewUrl(): string|null
+	public function previewUrl(VersionId|string $version = 'latest'): string|null
 	{
-		$preview = $this->blueprint()->preview();
+		$versionId = VersionId::from($version);
+		$url       = $this->blueprint()->preview();
 
-		if ($preview === false) {
+		if ($url === false) {
 			return null;
 		}
 
-		$url = match ($preview) {
-			true    => $this->url(),
-			default => $preview
+		$url = match ($url) {
+			true, null => $this->url(),
+			default    => $url
 		};
 
+		$uri = new Uri($url ?? $this->url());
```

Comment:
I don't think it needs the fallback as after the match statement, there is no way `$url` is `null`, right?